docs/_snippets/assets.js

```diff
@@ -85,7 +85,7 @@ window.attachTourBalloon = function( { target, text, tippyOptions } ) {
 		tooltip.hide();
 	} );
 
-	target.addEventListener( 'click', () => {
+	target.offsetParent.addEventListener( 'click', () => {
```

Comment:
This approach is incorrect. 

An offset parent is ([via MDN](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetParent)):

> a reference to the element which is the closest (nearest in the containment hierarchy) positioned ancestor element. If there is no positioned ancestor element, the nearest ancestor `td`, `th`, `table` will be returned, or the `body` if there are no ancestor table elements either.

so for a toolbar button in:

*   a `ClassicEditor`: this will be the outermost box of the editor. This is why it **looks** like it works because this box also contains the editing root but the tooltip will also disappear when you click any other button of the toolbar or an empty space in the toolbar.
*   an `InlineEditor` or a `BalloonEditor`: this will be the `.ck-balloon-panel` that hosts the toolbar. Focusing the editing root will not fire `click` on this element.
*   a `DecoupledDocumentEditor`: this will be the `div.main__content-inner` of the documentation page, which means clicking basically anywhere in the guide **will hide all tour balloons**.

There's no way to find out which editing root is related to the `target` so you need to pass the editor instance reference to this helper and create `once()` listener such as this to handle this properly

```javascript
editor.ui.view.editable.once( 'change:isFocused', ( evt, name, isFocused ) => {
    if ( isFocused ) {
        tooltip.hide();
    }
} );
```

Reply:
Edit: Iterating over `editor.editing.view.document.roots` and attaching a listener for each one of them will be safer since some editors such as multi-root may not have `editor.ui.view.editable`.

Reply:
I wanted to avoid passing `editor` and "tooltip will also disappear when you click any other button of the toolbar" is a feature IMO. But other arguments are obviously right, I will correct this now.